lib/date/date.ex

```diff
@@ -1298,4 +1298,18 @@ defmodule Timex.Date do
     {d,{h,m,s,round(u/1000)}}
   end
 
+  @doc """
+  Set the date to the beginning of the year
+  """
+  def beginning_of_year(%DateTime{:year => year, :timezone => tz}) do
+    from {year, 1, 1}, tz.full_name
+  end
+
+  @doc """
+  Set the date to the end of the year
```

Comment:
If possible, I'd like to use doctests for new API functions, because 1.) they provide better documentation for end users, and 2.) it is much easier to keep the tests and code in sync when they are next to each other. You can always provide both sets of tests (and I would encourage that too if you have functions with many possible variants/edge cases).